test/unit/registry-routes-components.js

```diff
@@ -104,6 +115,81 @@ describe('registry : routes : components', function(){
     });
   });
 
+  describe('when making valid info request for two components', function(){
+
+    before(function(done){
+      initialise(mockedComponents['async-error2-component']);
+      componentsRoute = new ComponentsRoute({}, mockedRepository);
+
+      makeInfoRequest({
+        components: [{
+          name: 'async-error2-component',
+          version: '1.X.X',
+          parameters: { error: true }
+        }, {
+          name: 'async-error2-component',
+          version: '1.0.0'
+        }]
+      }, done);
+    });
+
+    it('should return 200 status code', function(){
+      expect(code).to.be.equal(200);
+    });
+
+    it('should return a response containing both components', function(){
+      expect(response.length).to.be.equal(2);
+    });
+
+    it('should return a response containing components in the correct order', function(){
+      expect(response[0].response.href).to.be.undefined;
+      expect(response[1].response.href).to.be.equal('http://components.com/async-error2-component/1.0.0');
+    });
+
+    it('should return a response with error code and description for the first component', function(){
+      expect(response[0].response.code).to.be.equal('GENERIC_ERROR');
+      expect(response[0].response.error).to.be.equal('Component execution error: thisDoesnotExist is not defined');
```

Comment:
As previous comment says, I actually am not sure we want this here. I think info should just skip execution at all and give us just the versions. What do you think?